app/controllers/split_checkout_controller.rb

```diff
@@ -150,6 +154,31 @@ def update_order
     @order.errors.empty?
   end
 
+  def use_shipping_address_from_distributor
+    @order.ship_address = @order.address_from_distributor
+
+    # Add the missing data
+    bill_address = params[:order][:bill_address_attributes]
+    @order.ship_address.firstname = bill_address[:firstname]
+    @order.ship_address.lastname = bill_address[:lastname]
+    @order.ship_address.phone = bill_address[:phone]
+
+    # Remove shipping address from parameter so we don't override the address we just set
+    params[:order].delete(:ship_address_attributes)
+  end
+
+  def shipping_method_require_ship_address
+    return nil if allowed_shipping_methods.nil?
+
+    selected_shipping_method = allowed_shipping_methods.select do |sm|
+      sm.id.to_s == params[:shipping_method_id]
+    end
+
+    return nil if selected_shipping_method.empty?
+
+    selected_shipping_method.first.require_ship_address
+  end
+
```

Comment:
Above, we had to test the result of this with `== false`, which at first glance looks wrong. It's because we're returning one of three values: true, false or nil. ( Usual `nil` is treated the same as false).

I'd like to see if we can encapsulate the complexity a bit more and return a simple true/false value. 
Can we get away with this suggestion I wonder? It uses the `&` operator which lets you attempt a method only if the object is present (I forget what it's called..)

```suggestion
  def shipping_method_require_ship_address
    selected_shipping_method = allowed_shipping_methods&.select do |sm|
      sm.id.to_s == params[:shipping_method_id]
    end

    selected_shipping_method&.first&.require_ship_address == true
  end

```

Reply:
I see what you are trying to do, the proposed changes doesn't quite work. Would something like this be better ?

```ruby
  def shipping_method_ship_address_not_required?
    selected_shipping_method = allowed_shipping_methods&.select do |sm|
      sm.id.to_s == params[:shipping_method_id]
    end

    return false if selected_shipping_method.empty?

    selected_shipping_method.first.require_ship_address == false
  end

```
I kept a return as a feel like something like `selected_shipping_method&.first&.require_ship_address == false` is not easy to read, it's easy to miss the use of the Safe Navigation Operator `&.`

Reply:
👍 Looks good